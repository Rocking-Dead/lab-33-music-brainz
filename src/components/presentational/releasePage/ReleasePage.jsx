/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import Link from 'react-router-dom';

const ReleasePage = ({ releases }) => {
  const releaseList = releases.map(release => {
    return (
      <li key={release.id}>
        <Link to={`/releases/recordings/${release.id}`}>
          <figure>
            <img src={release.image} alt={`${release.title} cover image`} style={{ width: '100px' }} />
            <figcaption>
              {release.title}
            </figcaption>
          </figure>
        </Link>
      </li>
    );
  });
  return (
    <ul>
      { releaseList }
    </ul>
  );
};

ReleasePage.propTypes = {
  releases: PropTypes.arrayOf(PropTypes.shape({
    title:PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }
  ))
};

export default ReleasePage;
