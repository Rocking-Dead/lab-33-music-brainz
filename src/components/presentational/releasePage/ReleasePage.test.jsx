import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReleasePage from '../ReleasePage/ReleasePage';

describe('ReleasePage component', () => {
  afterEach(() => cleanup());
  it('renders ReleasePage', () => {
    const { asFragment } = render(<ReleasePage releases={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
