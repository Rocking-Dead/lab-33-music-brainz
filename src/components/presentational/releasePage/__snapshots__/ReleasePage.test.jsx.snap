// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleasePage component renders ReleasePage 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;
